include/message_filters/subscriber.h

```diff
@@ -39,74 +39,49 @@
 #include <type_traits>
 
 #include <rclcpp/rclcpp.hpp>
+#include <rclcpp/node_interfaces/node_interfaces.hpp>
+#include <rclcpp/create_subscription.hpp>
 
 #include "message_filters/connection.h"
 #include "message_filters/simple_filter.h"
 
 namespace message_filters
 {
 
-template<class NodeType = rclcpp::Node>
 class SubscriberBase
 {
 public:
-  typedef std::shared_ptr<NodeType> NodePtr;
+  using NodeParametersInterface = rclcpp::node_interfaces::NodeParametersInterface;
+  using NodeTopicsInterface = rclcpp::node_interfaces::NodeTopicsInterface;
 
-  virtual ~SubscriberBase() = default;
-  /**
-   * \brief Subscribe to a topic.
-   *
-   * If this Subscriber is already subscribed to a topic, this function will first unsubscribe.
-   *
-   * \param node The rclcpp::Node::SharedPtr to use to subscribe.
-   * \param topic The topic to subscribe to.
-   * \param qos (optional) The rmw qos profile to use to subscribe
-   */
-  virtual void subscribe(NodePtr node, const std::string& topic, const rmw_qos_profile_t qos = rmw_qos_profile_default) = 0;
+  using RequiredInterfaces = rclcpp::node_interfaces::NodeInterfaces<NodeParametersInterface, NodeTopicsInterface>;
+  using RequiredInterfacesPtr = std::shared_ptr<RequiredInterfaces>;
 
+  virtual ~SubscriberBase() = default;
   /**
    * \brief Subscribe to a topic.
    *
    * If this Subscriber is already subscribed to a topic, this function will first unsubscribe.
    *
-   * \param node The rclcpp::Node to use to subscribe.
+   * \param node The NodeInterfaces to use to subscribe.
    * \param topic The topic to subscribe to.
    * \param qos (optional) The rmw qos profile to use to subscribe
    */
-  virtual void subscribe(NodeType * node, const std::string& topic, const rmw_qos_profile_t qos = rmw_qos_profile_default) = 0;
```

Comment:
why ditching all subscribe functions?
I guess, this is unrelated to the actual change